proposals/spi-proposal.md

```diff
@@ -75,32 +79,31 @@ them separately by not specifying a CS pin and instead using Firmata
 `DIGITAL_MESSAGE` to control the CS pin state. If a CS pin is specified, the
 `csPinOptions` for that pin must also be specified.
 
-
 ```
 0:  START_SYSEX
 1:  SPI_DATA              (0x68)
-2:  SPI_BEGIN_TRANSACTION (0x01)
-3:  deviceId | channel    (bits 2-6: deviceId, bits 0-1: channel)
+2:  SPI_DEVICE_CONFIG     (0x01)
+3:  deviceId | channel    (bits 3-6: deviceId, bits 0-2: channel)
 4:  dataMode | bitOrder   (bits 1-2: dataMode (0-3), bit 0: bitOrder)
 5:  maxSpeed              (bits 0 - 6)
 6:  maxSpeed              (bits 7 - 14)
 7:  maxSpeed              (bits 15 - 21)
 8:  maxSpeed              (bits 22 - 28)
 9:  maxSpeed              (bits 29 - 32)
 10: wordSize              (0 = DEFAULT = 8-bits, 1 = 1-bit, 2 = 2 bits, etc)
-11: csPin                 [optional] (0-127) The chip select pin number
-12: csPinOptions          (required if csPin specified)
-                          bit 0: CS_ACTIVE_STATE (0 = Active LOW (default)
+11: csPinOptions          bit 0: CS_PIN_CONTROL (0 = disable
```

Comment:
Are these 2 states (disable | enable) sufficient or do we need additional states? For example in the Linux case would `disable` here mean ignore the csPin parameter and would `enable` always require the user to specify a valid csPin number. That later case is where I think there may be a need for an additional state such as:

- 0: disable // user will control pin manually
- 1: enable-specify-pin // auto control of pin and pin number required
- 2: enable-no-pin // auto control of pin but pin number not required

Reply:
If character special device files as described [here](https://www.kernel.org/doc/Documentation/spi/spidev) are used to access SPI devices on Linux the csPin parameter is of no relevance. The SPI driver know automatically which pin to used based on the file being accessed (`/dev/spidev<channel>.<deviceId>`). This implies that the CS_PIN_CONTROL option is also of no relevance on Linux.

Reply:
It sound like a binary enable/disable value for csPinOptions would be sufficient then and Linux implementations would simply ignore `csPin` and `csPinOptions`.